src/dynamic_boundary_conditions/hirds_depth_data_to_db.py

```diff
@@ -5,113 +5,144 @@
+    rainfall_data = pd.read_csv(filepath, skiprows=skip_rows, nrows=12)
+    rainfall_data.insert(0, "site_id", site_id)
+    rainfall_data.insert(1, "rcp", rcp)
+    rainfall_data.insert(2, "time_period", time_period)
+    rainfall_data.columns = rainfall_data.columns.str.lower()
+    return rainfall_data
+
+
+def add_rain_depth_data_to_db(engine, site_id: str, path):
+    """Store each site's rainfall data in the database. Each csv file contains historical and future forecasted
+    rainfall data for various rcp, time periods and durations.
+    To view the csv file, go to : https://hirds.niwa.co.nz/, select a site and generate a report,
+    there are rainfall depths and rainfall intensities data for different RCP Scenarios.
     To understand the structure of the CSV file, download the spreadsheet.
     """
-    the_values = [
-        (None, None, 12),
-        (2.6, "2031-2050", 40),
-        (2.6, "2081-2100", 54),
-        (4.5, "2031-2050", 68),
-        (4.5, "2081-2100", 82),
-        (6, "2031-2050", 96),
-        (6, "2081-2100", 110),
-        (8.5, "2031-2050", 124),
-        (8.5, "2081-2100", 138),
-    ]
-    filename = fr"{path}\{site_id}_depth.csv"
-    print("Adding data for site", site_id, "to database")
-    for (rcp, time_period, n) in the_values:
-        site_data = get_data_from_csv(
-            filename, site_id, rcp=rcp, time_period=time_period, n=n
-        )
-        site_data.to_sql("hirds_rain_depth", engine, index=False, if_exists="append")
-
-
-def hirds_depths_to_db(engine, catchment_area: Polygon, path):
-    """Store depth data of all the sites within the catchment area in the database."""
-    sites_in_catchment = get_sites_in_catchment(catchment_area, engine)
-    if check_table_exists(engine):
-        site_ids = get_sites_not_in_db(engine, sites_in_catchment)
-        if site_ids.size:
-            for site_id in site_ids:
-                rain_depth_data_from_hirds.get_data_from_hirds(site_id, path)
-                add_hirds_depth_data_to_db(path, site_id, engine)
+    filename = pathlib.Path(f"{site_id}_rain_depth.csv")
+    filepath = (path / filename)
+
+    layout_structure = get_layout_structure_of_csv(filepath)
+
+    for (skip_rows, rcp, time_period) in layout_structure:
+        site_data = get_data_from_csv(filepath, site_id, skip_rows=skip_rows, rcp=rcp, time_period=time_period)
+        site_data.to_sql("rainfall_depth", engine, index=False, if_exists="append")
+    log.info(f"Added rainfall depth data for site {site_id} to database")
+
+
+def add_each_site_rain_depth_data(engine, sites_id_list: list, path: str, idf: bool):
+    """Loop through all the sites in the sites list, download and store each site's rainfall data as a CSV file
+    in the desired path, and finally store it in the database."""
+    for site_id in sites_id_list:
+        rain_depth_data_from_hirds.store_data_to_csv(site_id, path, idf)
+        add_rain_depth_data_to_db(engine, site_id, path)
```

Comment:
Does the data need to be saved in a csv if it is also saved in the database?

Reply:
I initially thought about this too, but there might be a reason behind Pooja saving them as CSVs so I keep it like that. Let's have a short discussion about it. If this needs to be altered, I think it might be a good idea to do it in a separate branch as I think I will need to restructure some of the codes.

Reply:
We've briefly talked in person, I think this will need a bigger discussion. Let's do this at the end of the PR.

Reply:
You can always leave for now and address later, but I'm happy with whatever decision you two come too. I guess it's worth addressing now if leaving it for now will make more work in the future.

Reply:
Opened new issue #57. This is not urgent, but would be worth coming back to